src/EditorFeatures/VisualBasic/LineCommit/CommitFormatter.vb

```diff
@@ -52,7 +52,10 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
                 spanToFormat = spanToFormat.TranslateTo(currentSnapshot, SpanTrackingMode.EdgeInclusive)
                 dirtyRegion = dirtyRegion.TranslateTo(currentSnapshot, SpanTrackingMode.EdgeInclusive)
 
-                Dim document = currentSnapshot.GetOpenDocumentInCurrentContextWithChanges()
+                ' Use frozen partial semantics here.  We're operating on the UI thread, and we don't want to block the
+                ' user indefinitely while getting full semantics for this projects (which can require building all
+                ' projects we depend on).
+                Dim document = currentSnapshot.AsText().GetDocumentWithFrozenPartialSemantics(cancellationToken)
```

Comment:
How does `GetDocumentWithFrozenPartialSemantics` work? Return a partial solution with what currently available?

Reply:
Correct. It ensures the solution contains the current file contents, but it does not ensure metadata references have loaded, or that projects it depends on have up to date compilations either. It's good for when you need to do something synchronously on the ui thread and need semantics of some form.